sympy/sets/fancysets.py

```diff
@@ -352,112 +386,88 @@ def _is_multivariate(self):
         return len(self.lamda.variables) > 1
 
     def _contains(self, other):
-        from sympy.matrices import Matrix
+        from sympy.core.logic import fuzzy_and
         from sympy.solvers.solveset import solveset, linsolve
-        from sympy.solvers.solvers import solve
-        from sympy.utilities.iterables import is_sequence, cartes
-        L = self.lamda
-        if is_sequence(other) != is_sequence(L.expr):
-            return False
-        elif is_sequence(other) and len(L.expr) != len(other):
+
+        other = _sympify(other)
+        expr = self.lamda.expr
+        sig = self.lamda.signature
+        variables = self.lamda.variables
+
+        rep = {v: Dummy(v.name) for v in variables}
+        variables = [v.subs(rep) for v in variables]
```

Comment:
replace with a tuple instead of a list

Reply:
I guess this could be a generator expression since variables is only used once after. In fact the only thing done is to pass variables to set() so perhaps it would make sense to do that directly here: `{v.subs(rep) for c in variables}`.